addons/mail_bot/models/res_users.py

```diff
@@ -25,12 +26,17 @@ class ResUsers(models.Model):
     def SELF_READABLE_FIELDS(self):
         return super().SELF_READABLE_FIELDS + ['odoobot_state']
 
-    def _init_messaging(self, store):
+    @api.model
+    def _on_webclient_bootstrap(self):
+        if self.odoobot_state in [False, 'not_initialized'] and self.env.user and self.env.user._is_internal():
+            self.env.user._init_odoobot()
```

Comment:
`@api.model` is a decorator for rpc, it makes that the method can be called without a list of ids (i.e. with an empty recordset). Since this method is private, it cannot be called via rpc, the decorator is useless.

You call this method with an empty recordset, `self.odoobot_state` is always `False`

Reply:
If this is intended, I would rather read `if not self or ...`

Reply:
I remove the useless decorator.

> If this is intended, I would rather read if not self or ...

I don't understand this part, could you please elaborate?

Reply:
It is a common (arguably *bad*) practice to write:

```
if record.field == False```
to check *at the same time* that either the record is empty, either that the record exists but as field's value `False`.

In your code, since `self` is an empty recordset (called as `request.env['res.users']._on_webclient_bootstrap()`, i.e. without records) I *think* that by `if self.odoobot_state in [False, 'not_initialized']` you actually mean `if not self or self.odoobot_state == 'not_initialized'`. If it is the case (i.e. `False` isn't a valid value for the `odoobot_field`) then I would prefer we write `if not self or ...` than `if not self.odoobot_state in [False, ...]`.

Reply:
I see now, thank you.
I have updated the code accordingly.

Reply:
Or (imo better) we could make it required, with `not_initialized` be the default value